### PR TITLE
SAK-49515 Elfinder: Calculating the folder size in large sites cause severe performance issues

### PR DESCRIPTION
**Pull request by mpellicer**

### CODE REVIEW

File: kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/ContentToolFsVolumeFactory.java [hunk 1]
```diff
@@ -350,6 +352,11 @@ public long getSize(SakaiFsItem fsi) {
             String id = fsi.getId();
             try {
                 if (contentHostingService.isCollection(id)) {
+                    // In sites with many resources this causes severe performance issues.
+                    boolean showFolderSize = serverConfigurationService.getBoolean("elfinder.calculate.folder.sizes", true);
+                    if (!showFolderSize) {
+                        return 0;
+                    }
                     return contentHostingService.getCollectionSize(id);
```
Comment (bgarciaentornos) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/ContentToolFsVolumeFactory.java:360]:
```suggestion
                    return (showFolderSize) ? contentHostingService.getCollectionSize(id) : 0;
```
What do you think of this?

File: kernel/kernel-impl/src/main/webapp/WEB-INF/content-components.xml [hunk 2]
```diff
@@ -282,6 +283,7 @@
         <property name="sakaiFsService" ref="org.sakaiproject.elfinder.SakaiFsService"/>
         <property name="siteService" ref="org.sakaiproject.site.api.SiteService"/>
         <property name="securityService" ref="org.sakaiproject.authz.api.SecurityService"/>
+        <property name="serverConfigurationService" ref="org.sakaiproject.component.api.ServerConfigurationService"/>
```
Comment (bgarciaentornos) [inline kernel/kernel-impl/src/main/webapp/WEB-INF/content-components.xml:286]: This seems to belong to DropboxToolFsVolumeFactory but I don't see that file as updated. Is there anything missing or can we remove this?

**Review by bgarciaentornos — CHANGES_REQUESTED** [review]

File: kernel/kernel-impl/src/main/webapp/WEB-INF/content-components.xml (hunk 2 above)
Comment (mpellicer) [reply kernel/kernel-impl/src/main/webapp/WEB-INF/content-components.xml:286]: https://github.com/sakaiproject/sakai/blob/20e7d2bc2128d7fddbae19374d71f51ecf07aa1c/kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/DropboxToolFsVolumeFactory.java#L32

**Comment by mpellicer** [comment]
Thanks for the review Bernie @bgarciaentornos

**Pull request merged**